pkg/rescheduler/strategies/duplicates.go

```diff
@@ -21,49 +21,60 @@ import (
 	"strings"
 
 	"k8s.io/kubernetes/pkg/api/v1"
+	//TODO: Change to client-go instead of generated clientset.
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 
 	"github.com/aveshagarwal/rescheduler/pkg/rescheduler/evictions"
 	podutil "github.com/aveshagarwal/rescheduler/pkg/rescheduler/pod"
 )
 
+// This strategy evicts all duplicate pods on node. A pod is said to be a duplicate of other if
+// both of them have same name, kind and are within the same namespace. As of now, this strategy won't
+// evict daemonsets, mirror pods and pods with local storages.
+
 //type creator string
-type DuplicatePodsMap map[string][]*v1.Pod
+type duplicatePodsMap map[string][]*v1.Pod
 
-func RemoveDuplicatePods(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node) error {
+// RemoveDuplicatePods removes the duplicate pods on node.
+func RemoveDuplicatePods(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node) (int, error) {
+	podsEvicted := 0
 	for _, node := range nodes {
 		fmt.Printf("\nProcessing node: %#v\n", node.Name)
-		dpm := RemoveDuplicatePodsOnANode(client, node)
+		dpm := listDuplicatePodsOnANode(client, node)
 		for creator, pods := range dpm {
 			if len(pods) > 1 {
+				fmt.Println(len(pods))
 				fmt.Printf("%#v\n", creator)
 				// i = 0 does not evict the first pod
 				for i := 1; i < len(pods); i++ {
 					//fmt.Printf("Removing duplicate pod %#v\n", k.Name)
 					success, err := evictions.EvictPod(client, pods[i], policyGroupVersion)
 					if !success {
+						//TODO: change fmt.Printf as glogs.
 						fmt.Printf("Error when evicting pod: %#v (%#v)\n", pods[i].Name, err)
 					} else {
+						podsEvicted++
```

Comment:
why do you need this, even for testing?

Reply:
> why do you need this, even for testing?

So as of now, we are silently failing for the pods that are not getting deleted but I thought we need to collect the errors and as well as pods that failed. This will be helpful for debugging later.

Reply:
That is not the case, as its not about silently falling.. Pods not getting deleted is not an error always, It could be due to Pod Disruption Budget (PDB) too. Also rescheduler is best effort, that means that it just tries to evict a pod, but does not force.  Though, it would be good to know why a pod was not evicted, not sure how returning number of evicted pods is useful. Anyway let me think about it more and will let you know after you update. thanks.

Reply:
I agree that a number will not be useful but a map with podName and error may be helpful is what I meant. I have rebased the branch so as to avoid conflicts.

Reply:
I think It'd be better to have behind some glog levels once we move to glog from fmt. So only during debugging, it is enabled not by default.

Reply:
Avesh, PTAL. I created another function which does the same thing but get called from RemoveDuplicatePods  and RemoveDuplicatePods won't return anything.